src/kbmod/search/kernels/kernels.cu

```diff
@@ -220,6 +220,16 @@ __global__ void searchFilterImages(PsiPhiArrayMeta psi_phi_meta, void *psi_phi_v
     assert(psi_phi_vect != nullptr && image_times != nullptr && trajectories != nullptr &&
            results != nullptr);
 
+    // Copy the times to faster shared memory for the block. Make sure all threads
+    // copy their time before progressing. We need to do this before pruning on
+    // (x, y) in order to correctly handle blocks at the edge of the image.
+    __shared__ double shared_times[MAX_NUM_IMAGES];
+    int time_idx = threadIdx.x + threadIdx.y * THREAD_DIM_X;
```

Comment:
I believe you that this is the correct calculation of `time_idx`, but I not familiar enough to say with certainty. Is it really `threadIdx.x + threadIdx.y * THREAD_DIM_X`?

Reply:
For the current code, I think that will work.  But I changed it to:

int time_idx = threadIdx.x + threadIdx.y * blockDim.x;

which should be safer in case the code that calls the kernel function changes.